File: molechess/src/main/java/org/chernovia/molechess/MoleResult.java
package org.chernovia.molechess;

public class MoleResult {
  boolean success;
  
  String message;
  
  public MoleResult(String s) {
    this.success = true;
    this.message = s;
  }
  
  public MoleResult(boolean b, String s) {
    this.success = b;
    this.message = s;
  }
}

File: src/main/kotlin/org/chernovia/molechess/MoleResult.kt
package org.chernovia.molechess

data class MoleResult @JvmOverloads constructor(
    @JvmField val success: Boolean = true,
    @JvmField val message: String?,
    @JvmField val user: MoleUser? = null,
    @JvmField val player: MolePlayer? = null
)